core/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceAggregatorFactory.java

```diff
@@ -47,11 +69,92 @@ public AggregatorFactory subFactories(AggregatorFactories subFactories) {
         }
     }
 
+    private final Class<VS> valuesSourceType;
+    private final ValueType targetValueType;
+    private String field = null;
+    private Script script = null;
+    private ValueType valueType = null;
+    private String format = null;
+    private Object missing = null;
+    private DateTimeZone timeZone;
     protected ValuesSourceConfig<VS> config;
 
-    protected ValuesSourceAggregatorFactory(String name, String type, ValuesSourceConfig<VS> config) {
+    // NORELEASE remove this method when aggs refactoring complete
+    /**
+     * This constructor remains here until all subclasses have been moved to the
+     * new constructor. This also means moving from using
+     * {@link ValuesSourceParser} to using {@link AbstractValuesSourceParser}.
+     */
+    @Deprecated
+    protected ValuesSourceAggregatorFactory(String name, String type, ValuesSourceParser.Input<VS> input) {
         super(name, type);
-        this.config = config;
+        this.valuesSourceType = input.valuesSourceType;
+        this.targetValueType = input.targetValueType;
+        this.field = input.field;
+        this.script = input.script;
+        this.valueType = input.valueType;
+        this.format = input.format;
+        this.missing = input.missing;
+        this.timeZone = input.timezone;
+    }
+
+    protected ValuesSourceAggregatorFactory(String name, String type, Class<VS> valuesSourceType, ValueType targetValueType) {
+        super(name, type);
+        this.valuesSourceType = valuesSourceType;
+        this.targetValueType = targetValueType;
+    }
+
+    public void field(String field) {
+        this.field = field;
+    }
+
+    public String field() {
+        return field;
+    }
+
+    public void script(Script script) {
+        this.script = script;
+    }
+
+    public Script script() {
+        return script;
+    }
+
+    public void valueType(ValueType valueType) {
+        this.valueType = valueType;
+    }
+
+    public void format(String format) {
+        this.format = format;
+    }
+
+    public String format() {
+        return format;
+    }
+
+    public void missing(Object missing) {
+        this.missing = missing;
+    }
+
+    public Object missing() {
+        return missing;
+    }
+
+    public void timeZone(DateTimeZone timeZone) {
+        this.timeZone = timeZone;
+    }
+
+    public DateTimeZone timeZone() {
+        return timeZone;
+    }
```

Comment:
if I understand correctly, this class will ultimately be used in the client to create aggs, so could you add some javadocs to the above methods?